text-serializer-ansi/src/test/java/net/kyori/adventure/text/serializer/ansi/AnsiComponentSerializerTest.java

```diff
@@ -0,0 +1,190 @@
+        TextComponent.of("bar")
+          .color(NamedTextColor.BLUE)
+      )
+      .append(TextComponent.of("baz"))
+      .build();
+    final String expected = Ansi.ansi()
+      .a(Ansi.Attribute.UNDERLINE_DOUBLE)
+      .a("hi").a(Ansi.Attribute.RESET)
+      .a(Ansi.ansi().fgBright(Ansi.Color.GREEN))
+      .a("foo").a(Ansi.Attribute.RESET)
+      .a(Ansi.ansi().fgBright(Ansi.Color.BLUE))
+      .a(Ansi.Attribute.UNDERLINE_DOUBLE)
+      .a("bar").a(Ansi.ansi().a(Ansi.Attribute.RESET))
+      .a(Ansi.Attribute.UNDERLINE_DOUBLE)
+      .a("baz").a(Ansi.Attribute.RESET)
+      .toString();
+    assertEquals(expected, AnsiComponentSerializer.fullColour().serialize(c1));
+
+    final TextComponent c2 = TextComponent.builder()
+      .content("")
+      .color(NamedTextColor.YELLOW)
+      .append(TextComponent.builder()
+        .content("Hello ")
+        .append(
+          TextComponent.builder()
+            .content("world")
+            .color(NamedTextColor.GREEN)
+            .build()
+        )
+        .append(TextComponent.of("!")) // Should be yellow
+        .build()
+      )
+      .build();
+    final Ansi ansi = Ansi.ansi()
+      .fgBright(Ansi.Color.YELLOW)
+      .a("Hello ").a(Ansi.ansi().a(Ansi.Attribute.RESET))
+      .fgBright(Ansi.Color.GREEN)
+      .a("world").a(Ansi.ansi().a(Ansi.Attribute.RESET))
+      .fgBright(Ansi.Color.YELLOW)
+      .a("!").reset();
+    assertEquals(ansi.toString(), AnsiComponentSerializer.fullColour().serialize(c2));
+
+    final TextComponent c3 = TextComponent.builder()
+      .content("")
+      .decoration(TextDecoration.BOLD, true)
+      .append(
+        TextComponent.builder()
+          .content("")
+          .color(NamedTextColor.YELLOW)
+          .append(TextComponent.builder()
+            .content("Hello ")
+            .append(
+              TextComponent.builder()
+                .content("world")
+                .color(NamedTextColor.GREEN)
+                .build()
+            )
+            .append(TextComponent.of("!"))
+            .build()
+          )
+          .build())
+      .build();
+    final Ansi expected3 = Ansi.ansi()
+      .fgBright(Ansi.Color.YELLOW).a(Ansi.ansi().a(Ansi.Attribute.UNDERLINE_DOUBLE))
+      .a("Hello ").a(Ansi.ansi().a(Ansi.Attribute.RESET))
+      .fgBright(Ansi.Color.GREEN).a(Ansi.ansi().a(Ansi.Attribute.UNDERLINE_DOUBLE))
+      .a("world").a(Ansi.ansi().a(Ansi.Attribute.RESET))
+      .fgBright(Ansi.Color.YELLOW).a(Ansi.ansi().a(Ansi.Attribute.UNDERLINE_DOUBLE))
+      .a("!").reset();
+    assertEquals(expected3.toString(), AnsiComponentSerializer.fullColour().serialize(c3));
+  }
+
+  @Test
+  void testHexColor(){
+    final TextColor color = TextColor.of(0x77ff33);
+    final TextComponent c0 = TextComponent.of("Kittens!", color);
+    final int red = color.red();
+    final int blue = color.blue();
+    final int green = color.green();
+    final Ansi ansiColor = Ansi.ansi().format("\u001b[38;2;%d;%d;%dm", red, green, blue);
```

Comment:
didn't you make a constant for this in `AnsiComponentSerializerImpl`?

Reply:
Yes but I have never use my working constants in a test - it implies that if you change it up in the code base a test may still pass even though its incorrect.